src/shared/index.ts

```diff
@@ -141,6 +141,21 @@ export function augmentLog(log: Log, rules?: Map<string, ReportingDescriptor>) {
     log._distinct = mapDistinct(fileAndUris);
 }
 
+export function effectiveLevel(result: Result): Result.level {
+    switch (result.kind) {
+        case 'informational':
+        case 'notApplicable':
+        case 'pass':
+            return 'note';
+        case 'open':
+        case 'review':
+            return 'warning';
+        case 'fail':
+        default:    // default case should never happen.
```

Comment:
Confused by why "default case should never happen". I often seen `kind` undefined.

Reply:
I may be wrong about that comment.

In the .NET SDK, the deserialization is controlled by a set of property-level attributes such that if `Kind` is missing, it will populate `fail` into the object model (it's a bit more complicated, but that's the idea). So in .NET there will always be a valid value. Not so here, I take it.

---
In reply to: [493825383](https://github.com/microsoft/sarif-vscode-extension/pull/359#discussion_r493825383) [](ancestors = 493825383)